### PR TITLE
Netmiko

### PR DESCRIPTION
**Pull request by TerminalFin**
Initial submission to uplift this content pack to officially supported status. I request assistance in developing the unit tests so that I can meet the XSOAR contribution standards.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [X] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [X] Documentation

### CODE REVIEW

**Comment by merit-maita** [comment]
@ShacharKidor FYI

**Review by dantavori — CHANGES_REQUESTED** [review]
hey, thank you for your contribution :)
Feel free to reach me on Slack for anything.

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.yml [hunk 1]
```diff
@@ -3,304 +3,492 @@ commonfields:
...
-      name: override_host
-    - description: If specified, it will override the target port specified in the
-        integration instrance.
-      name: override_port
-    - auto: PREDEFINED
-      description: If specified, it will override the target platform specified in
-        the integration instrance.
-      name: override_platform
-      predefined:
+  - name: port
```
[the remark this answers is not in the document]
Comment (TerminalFin) [reply Packs/Netmiko/Integrations/Netmiko/Netmiko.yml:6]: I did not, this was done automatically by VS Code.

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.yml [hunk 2]
```diff
@@ -3,304 +3,492 @@ commonfields:
...
+  dockerimage: demisto/netmiko:1.0.0.55567
   runonce: false
-  script: ''
+  script: ""
   subtype: python3
   type: python
 fromversion: 6.0.0
 tests:
-- No tests (auto formatted)
+  - No tests (auto formatted)
```
[the remark this answers is not in the document]
Comment (TerminalFin) [reply Packs/Netmiko/Integrations/Netmiko/Netmiko.yml:494]: Added per our conversation on 5/1/2023

File: Packs/Netmiko/Integrations/Netmiko/Netmiko_description.md [hunk 3]
```diff
@@ -1,201 +1,231 @@
...
+- Updated content pack to uplift it to Palo Alto Networks officially supported status
+
 # Netmiko
-  Multi-vendor library to simplify SSH connections to network devices using the Netmiko Python library.
+Multi-vendor library to execute commands over SSH connections for network devices using the Netmiko Python library.
 
-This integration uses the Netmiko library to call Paramiko commands. This can help in situations where there is no integration for a particular vendor technology, but, has a CLI that can be interacted with over the network.
+This can help in situations where there is no integration for a particular vendor technology, but a CLI is accessible that can be interacted with over the network.
 
 ## Platforms:
```
[the remark this answers is not in the document]
Comment (TerminalFin) [reply Packs/Netmiko/Integrations/Netmiko/Netmiko_description.md:13]: Resolved

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py [hunk 4]
```diff
@@ -141,10 +141,11 @@ def cmds_command(client, args):
         outputs=outputs,
         readable_output=md
     )
-    return_results(command_results)
+    
+    return command_results
 
 
-def main():
+def main():     # pragma: no cover
```
Comment (dantavori) [inline Packs/Netmiko/Integrations/Netmiko/Netmiko.py:148]:
```suggestion
def main():  # pragma: no cover
```

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py [hunk 5]
```diff
@@ -126,25 +134,38 @@ def cmds_command(client, args):
             demisto.error(f"Error with raw print output - {err}")
 
     else:
-        md = tableToMarkdown(f'Command(s) against {client.hostname} ({client.platform}):', output.get('Commands', []))
+        hdrs = ["Hostname", "DateTimeUTC", "Command", "Output"]
+        data = []
+
+        # Single command
+        if len(cmds) == 1:
+            data.append(output["Commands"][0])
```
Comment (dantavori) [inline Packs/Netmiko/Integrations/Netmiko/Netmiko.py:142]: is `Commands` key guaranteed to exist under `outputs`?

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py [hunk 6]
```diff
@@ -39,7 +45,7 @@ def connect(self):
         else:
             try:
                 self.net_connect = Netmiko(device_type=self.platform, host=self.hostname, port=self.port,
-                                           use_keys=True, username=self.username, password=self.password)
+                                           use_keys=False, username=self.username, password=self.password)
```
Comment (dantavori) [inline Packs/Netmiko/Integrations/Netmiko/Netmiko.py:48]: what's the reason for this change?

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py [hunk 7]
```diff
@@ -82,6 +88,8 @@ def test_command(client):
 
 def cmds_command(client, args):
 
+    interestingKeys = ("Hostname", "DateTimeUTC", "Command", "Output")
```
Comment (dantavori) [inline Packs/Netmiko/Integrations/Netmiko/Netmiko.py:91]: let's use the `hdrs` variable below if it's exactly the same

**Review by dantavori — COMMENTED** [review]
nice,
see build failures:
1. validate [fails](https://app.circleci.com/pipelines/github/demisto/content/256960/workflows/abe784d8-0df3-44eb-b478-37db837e9a96/jobs/698998) on docker update (need to update in yml & release notes) and incomplete release notes (see note from the previous review)
2. lint [fails](https://app.circleci.com/pipelines/github/demisto/content/256960/workflows/abe784d8-0df3-44eb-b478-37db837e9a96/jobs/699000) on:
    - some flake8 errors
    - unit-tests coverage should be at least 70%. you can add the `pragma: no cover` comment in the client methods, it will increase the value.

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py (hunk 5 above)
Comment (TerminalFin) [reply Packs/Netmiko/Integrations/Netmiko/Netmiko.py:142]: Yes, Commands it the output returned after the cmds= commands are executed. In this case, only a single command was passed as an argument (ifconfig, whoami, etc.).

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py (hunk 7 above)
Comment (TerminalFin) [reply Packs/Netmiko/Integrations/Netmiko/Netmiko.py:91]: Done

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py (hunk 6 above)
Comment (TerminalFin) [reply Packs/Netmiko/Integrations/Netmiko/Netmiko.py:48]: use_keys was changed to False since we aren't passing an SSH Key Certificate as part of the credentials parameter. As such, we are simply using a standard username/password from the credential parameter.

File: Packs/Netmiko/Integrations/Netmiko/Netmiko.py [hunk 8]
```diff
@@ -1,25 +1,31 @@
...
 
 
-def return_file(keys):
+# Return only specific keys from dictionary
+def include_keys(dictionary, keys):  # pragma: no cover
+    key_set = set(keys) & set(dictionary.keys())
+    return {key: dictionary[key] for key in key_set}
+
+
+def return_file(keys):  # pragma: no cover
```
Comment (dantavori) [inline Packs/Netmiko/Integrations/Netmiko/Netmiko.py:23]: please avoid the comments when possible

**Review by dantavori — APPROVED** [review]
